libvast/src/address.cpp

```diff
@@ -21,6 +21,25 @@ address::address() {
   bytes_.fill(0);
 }
 
+address::address(uint8_t const* bytes, family fam, byte_order order) {
+  auto b = reinterpret_cast<const uint32_t*>(bytes);
+  if (fam == ipv4) {
+    std::copy(v4_mapped_prefix.begin(), v4_mapped_prefix.end(), bytes_.begin());
+    auto p = reinterpret_cast<uint32_t*>(&bytes_[12]);
+    if (order == host)
+      *p = detail::to_network_order(*b);
+    else
+      *p = *b;
```

Comment:
You can simplify this to this one-liner: `*p = order == host ? detail::to_network_order(*b) ? *b`.